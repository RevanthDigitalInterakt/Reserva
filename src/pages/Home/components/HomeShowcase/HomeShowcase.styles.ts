import { StyleSheet } from 'react-native';
import { FONTS } from '../../../../base/styles/fonts';

export const styles = StyleSheet.create({
  container: {
    marginVertical: 10,
  },
  shelfName: {
    fontFamily: FONTS.RESERVA_DISPLAY_REGULAR,
    fontSize: 24,
    lineHeight: 23,
    letterSpacing: 0,
  },
  shelfContainer: {
    marginTop: 10,
  },
  shelf: {
    alignItems: 'center',
    justifyContent: 'center',
  },
});
