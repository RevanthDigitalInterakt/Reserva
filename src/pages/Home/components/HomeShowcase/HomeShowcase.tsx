import React from 'react';
import {
  View, FlatList, Text, Image, TouchableOpacity,
} from 'react-native';
import { trackClickSmartHintStore } from '../../../../zustand/useTrackClickSmartHint/useTrackClickSmartHint';
import { TrackPageTypeEnum } from '../../../../base/graphql/generated';
import { styles } from './HomeShowcase.styles';
import { COLORS, FONTS } from '../../../../base/styles';
import { decimalPart, integerPart } from '../../../../utils/numberUtils';

interface IFlag {
  type: string;
  value?: number;
  text?: string;
}

interface ISize {
  value: string;
  disabled: boolean;
}

interface IPrice {
  listPrice: number;
  salePrice: number;
}

interface IColor {
  name: string;
  hex: string;
  disabled?: boolean;
  sizes: ISize[];
}

interface ISku {
  skuId: string;
  colors: IColor[];
}

interface IProduct {
  productName: string;
  productId: string;
  productLink: string;
  brand: string;
  image: string;
  categoryTree: string[];
  flags: IFlag[];
  sku: ISku[];
  prices: IPrice;
}

interface IShelf {
  shelfName: string;
  products: IProduct[];
}

export function HomeShowcase() {
  const data: IShelf[] = [
    {
      shelfName: 'Vitrine de teste da SmartHint estou testanto o tamanho dela',
      products: [
        {
          productName: 'Camisa Polo Reserva Linho Braco',
          productId: '1',
          productLink: 'https://usereserva.com/camisa/p',
          brand: 'RESERVA',
          image: 'https://lojausereserva.vtexassets.com/arquivos/ids/8409628-400-600',
          categoryTree: [
            'Camisas',
          ],
          flags: [
            {
              type: 'savings',
              value: 20,
            },
            {
              type: 'cashback',
              value: 15,
            },
          ],
          sku: [{
            skuId: '5678',
            colors: [
              {
                name: 'BRANCO',
                hex: '#FFF',
                disabled: false,
                sizes: [
                  {
                    value: 'P',
                    disabled: false,
                  },
                ],
              },
            ],
          }],
          prices: {
            listPrice: 600.90,
            salePrice: 400.50,
          },
        },
        {
          productName: 'Camisa Braca',
          productId: '2',
          productLink: 'https://usereserva.com/camisa/p',
          brand: 'RESERVA',
          image: 'https://lojausereserva.vtexassets.com/arquivos/ids/8409628-400-600',
          categoryTree: [
            'Camisas',
          ],
          flags: [
            {
              type: 'savings',
              value: 15,
            },
            {
              type: 'cashback',
              value: 10,
            },
          ],
          sku: [{
            skuId: '5679',
            colors: [
              {
                name: 'BRANCO',
                hex: '#FFF',
                disabled: false,
                sizes: [
                  {
                    value: 'P',
                    disabled: false,
                  },
                ],
              },
            ],
          }],
          prices: {
            listPrice: 600.00,
            salePrice: 400.00,
          },
        },
        {
          productName: 'Camisa Azul',
          productId: '3',
          productLink: 'https://usereserva.com/camisa/p',
          brand: 'RESERVA',
          image: 'https://lojausereserva.vtexassets.com/arquivos/ids/8409628-400-600',
          categoryTree: [
            'Camisas',
          ],
          flags: [
            {
              type: 'savings',
              value: 25,
            },
            {
              type: 'cashback',
              value: 15,
            },
          ],
          sku: [{
            skuId: '5679',
            colors: [
              {
                name: 'BRANCO',
                hex: '#FFF',
                disabled: false,
                sizes: [
                  {
                    value: 'P',
                    disabled: false,
                  },
                ],
              },
            ],
          }],
          prices: {
            listPrice: 600.00,
            salePrice: 400.00,
          },
        },
        {
          productName: 'Camisa Verde',
          productId: '4',
          productLink: 'https://usereserva.com/camisa/p',
          brand: 'RESERVA',
          image: 'https://lojausereserva.vtexassets.com/arquivos/ids/8409628-400-600',
          categoryTree: [
            'Camisas',
          ],
          flags: [
          ],
          sku: [{
            skuId: '5679',
            colors: [
              {
                name: 'BRANCO',
                hex: '#FFF',
                disabled: false,
                sizes: [
                  {
                    value: 'P',
                    disabled: false,
                  },
                ],
              },
            ],
          }],
          prices: {
            listPrice: 600.00,
            salePrice: 0,
          },
        },
      ],
    },
  ];

  const renderItem = ({ item: shelf }: { item: IShelf }) => (
    <View style={{ marginTop: 10 }}>
      <View style={{ alignItems: 'center', justifyContent: 'center' }}>
        <Text style={{
          fontFamily: FONTS.RESERVA_DISPLAY_REGULAR, fontSize: 24, lineHeight: 23, letterSpacing: 0,
        }}
        >
          {shelf.shelfName}
        </Text>
      </View>
      <FlatList
        horizontal
        data={shelf.products}
        keyExtractor={(item) => item.productId}
        renderItem={({ item: product }) => (
          <TouchableOpacity
            style={{ padding: 10, borderRadius: 10 }}
            onPress={() => {
              trackClickSmartHintStore.getState()
                .onSendTrackClick(product.productId, TrackPageTypeEnum.Home);
            }}
          >
            <Image
              source={{ uri: product.image }}
              style={{
                width: 155,
                height: 270,
                borderRadius: 10,
              }}
            />
            <Text style={{
              fontFamily: FONTS.RESERVA_SANS_BOLD,
              fontSize: 15,
              lineHeight: 20,
            }}
            >
              {product.productName.length > 22 ? `${product.productName.substring(0, 20).trim()}..` : product.productName}
            </Text>
            <View style={{ flexDirection: 'row', alignItems: 'center' }}>
              {product.prices.salePrice !== 0 ? (
                <>
                  <Text
                    style={{
                      fontSize: 17,
                      lineHeight: 23,
                      fontFamily: FONTS.RESERVA_SANS_BOLD,
                    }}
                  >
                    {`R$ ${integerPart(product.prices.salePrice || 0)}`}
                  </Text>
                  <Text
                    style={{
                      fontSize: 11,
                      fontFamily: FONTS.RESERVA_SANS_BOLD,
                      marginTop: -4,
                    }}
                  >
                    {`,${decimalPart(product.prices.salePrice || 0)}`}
                  </Text>
                </>
              ) : (
                <>
                  <Text
                    style={{
                      fontSize: 17,
                      lineHeight: 23,
                      fontFamily: FONTS.RESERVA_SANS_BOLD,
                    }}
                  >
                    {`R$ ${integerPart(product.prices.listPrice || 0)}`}
                  </Text>
                  <Text
                    style={{
                      fontSize: 11,
                      fontFamily: FONTS.RESERVA_SANS_BOLD,
                      marginTop: -4,
                    }}
                  >
                    {`,${decimalPart(product.prices.listPrice || 0)}`}
                  </Text>
                </>
              )}
              {product.prices.salePrice !== 0 ? (
                <>
                  <Text
                    style={{
                      marginLeft: 10,
                      color: COLORS.LIGHT_GRAY,
                      textDecorationLine: 'line-through',
                      fontFamily: FONTS.NUNITO_REGULAR,
                      fontSize: 17,
                      lineHeight: 23,
                    }}
                  >
                    {`${integerPart(product.prices.listPrice || 0)}`}
                  </Text>
                  <Text
                    style={{
                      color: COLORS.LIGHT_GRAY,
                      fontFamily: FONTS.NUNITO_REGULAR,
                      lineHeight: 17,
                      fontSize: 11,
                    }}
                  >
                    {`,${decimalPart(product.prices.listPrice || 0)}`}
                  </Text>
                </>
              ) : (
                null
              )}
            </View>
            {product.flags.map((flag) => {
              if (flag.type === 'savings') {
                return (
                  <View style={{
                    position: 'absolute',
                    alignItems: 'center',
                    justifyContent: 'center',
                    flexDirection: 'row',
                    top: 245,
                    left: 20,
                    backgroundColor: COLORS.BLACK,
                    width: 71,
                    height: 30,
                    borderRadius: 30,
                  }}
                  >
                    <Text
                      key={flag.type}
                      style={{
                        color: COLORS.WHITE,
                        fontFamily: FONTS.WORK_SANS_BOLD_ITALIC,
                        fontSize: 13,
                        lineHeight: 15,
                      }}
                    >
                      {`${flag.value}%`}
                    </Text>
                    <Text
                      style={{
                        color: COLORS.WHITE,
                        fontFamily: FONTS.WORK_SANS_ITALIC,
                        fontSize: 13,
                        lineHeight: 15,
                        letterSpacing: -1.5,
                        marginLeft: 5,
                      }}
                    >
                      OFF
                    </Text>
                  </View>
                );
              }
              if (flag.type === 'cashback') {
                return (
                  <View style={{
                    alignItems: 'center',
                    justifyContent: 'center',
                    backgroundColor: COLORS.LIGHT_GREY_BLUE,
                    width: 150,
                    height: 15,
                    borderRadius: 30,
                    flexDirection: 'row',
                  }}
                  >
                    <Text
                      key={flag.type}
                      style={{
                        color: COLORS.BLACK,
                        fontFamily: FONTS.WORK_SANS_BOLD,
                        fontSize: 12,
                        lineHeight: 15,
                      }}
                    >
                      {`Ganhe ${flag.value}%`}
                    </Text>
                    <Text
                      key={flag.type}
                      style={{
                        color: COLORS.BLACK,
                        fontFamily: FONTS.WORK_SANS_REGULAR,
                        fontSize: 12,
                        lineHeight: 15,
                      }}
                    >
                      {' de cashback'}
                    </Text>
                  </View>
                );
              }
              return null;
            })}

          </TouchableOpacity>
        )}
      />
    </View>
  );

  return (
    <View style={styles.conteiner}>
      <FlatList
        data={data}
        renderItem={renderItem}
        keyExtractor={(item, index) => item.shelfName + index.toString()}
      />
    </View>
  );
}
