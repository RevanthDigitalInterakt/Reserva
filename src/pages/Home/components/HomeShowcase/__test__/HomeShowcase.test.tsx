import React from 'react';
import { render, screen } from '@testing-library/react-native';
import { ThemeProvider } from 'styled-components/native';
import { MockedProvider } from '@apollo/client/testing';
import { theme } from '../../../../../base/usereservappLegacy/theme';
import { HomeShowcase } from '../HomeShowcase';

const TestingComponent = (
  <ThemeProvider theme={theme}>
    <MockedProvider addTypename={false}>
      <HomeShowcase />
    </MockedProvider>
  </ThemeProvider>
);

describe('HomeShowcase', () => {
  it('snapshot', () => {
    render(TestingComponent);

    expect(screen.toJSON()).toMatchSnapshot();
  });
});
