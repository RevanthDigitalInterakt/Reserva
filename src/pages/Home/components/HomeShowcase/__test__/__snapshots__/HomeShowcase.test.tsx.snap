// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeShowcase snapshot 1`] = `
<View
  style={
    {
      "marginVertical": 10,
    }
  }
>
  <RCTScrollView
    data={
      [
        {
          "products": [
            {
              "brand": "RESERVA",
              "categoryTree": [
                "Camisas",
              ],
              "flags": [
                {
                  "type": "savings",
                  "value": 20,
                },
                {
                  "type": "cashback",
                  "value": 15,
                },
              ],
              "image": "https://lojausereserva.vtexassets.com/arquivos/ids/8409628-400-600",
              "prices": {
                "listPrice": 600.9,
                "salePrice": 400.5,
              },
              "productId": "1",
              "productLink": "https://usereserva.com/camisa/p",
              "productName": "Camisa Polo Reserva Linho Braco",
              "sku": [
                {
                  "colorHex": "#FFF",
                  "colorName": "BRANCO",
                  "colorRefId": "5678",
                  "sizes": [
                    {
                      "disabled": false,
                      "skuId": "5678",
                      "value": "P",
                    },
                  ],
                },
              ],
            },
            {
              "brand": "RESERVA",
              "categoryTree": [
                "Camisas",
              ],
              "flags": [
                {
                  "type": "savings",
                  "value": 20,
                },
                {
                  "type": "cashback",
                  "value": 15,
                },
              ],
              "image": "https://lojausereserva.vtexassets.com/arquivos/ids/8409628-400-600",
              "prices": {
                "listPrice": 600.9,
                "salePrice": 400.5,
              },
              "productId": "1",
              "productLink": "https://usereserva.com/camisa/p",
              "productName": "Camisa Polo Reserva Linho Braco",
              "sku": [
                {
                  "colorHex": "#FFF",
                  "colorName": "BRANCO",
                  "colorRefId": "5678",
                  "sizes": [
                    {
                      "disabled": false,
                      "skuId": "5678",
                      "value": "P",
                    },
                  ],
                },
              ],
            },
            {
              "brand": "RESERVA",
              "categoryTree": [
                "Camisas",
              ],
              "flags": [
                {
                  "type": "savings",
                  "value": 20,
                },
              ],
              "image": "https://lojausereserva.vtexassets.com/arquivos/ids/8409628-400-600",
              "prices": {
                "listPrice": 600.9,
                "salePrice": 400.5,
              },
              "productId": "1",
              "productLink": "https://usereserva.com/camisa/p",
              "productName": "Camisa Polo Reserva Linho Braco",
              "sku": [
                {
                  "colorHex": "#FFF",
                  "colorName": "BRANCO",
                  "colorRefId": "5678",
                  "sizes": [
                    {
                      "disabled": false,
                      "skuId": "5678",
                      "value": "P",
                    },
                  ],
                },
              ],
            },
            {
              "brand": "RESERVA",
              "categoryTree": [
                "Camisas",
              ],
              "flags": [],
              "image": "https://lojausereserva.vtexassets.com/arquivos/ids/8409628-400-600",
              "prices": {
                "listPrice": 600.9,
                "salePrice": 0,
              },
              "productId": "1",
              "productLink": "https://usereserva.com/camisa/p",
              "productName": "Camisa Polo Reserva Linho Braco",
              "sku": [
                {
                  "colorHex": "#FFF",
                  "colorName": "BRANCO",
                  "colorRefId": "5678",
                  "sizes": [
                    {
                      "disabled": false,
                      "skuId": "5678",
                      "value": "P",
                    },
                  ],
                },
              ],
            },
          ],
          "shelfName": "Vitrine de teste da SmartHint estou testanto o tamanho dela",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "marginTop": 10,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                {
                  "fontFamily": "ReservaDisplay-Regular",
                  "fontSize": 24,
                  "letterSpacing": 0,
                  "lineHeight": 23,
                }
              }
            >
              Vitrine de teste da SmartHint estou testanto o tamanho dela
            </Text>
          </View>
          <RCTScrollView
            data={
              [
                {
                  "brand": "RESERVA",
                  "categoryTree": [
                    "Camisas",
                  ],
                  "flags": [
                    {
                      "type": "savings",
                      "value": 20,
                    },
                    {
                      "type": "cashback",
                      "value": 15,
                    },
                  ],
                  "image": "https://lojausereserva.vtexassets.com/arquivos/ids/8409628-400-600",
                  "prices": {
                    "listPrice": 600.9,
                    "salePrice": 400.5,
                  },
                  "productId": "1",
                  "productLink": "https://usereserva.com/camisa/p",
                  "productName": "Camisa Polo Reserva Linho Braco",
                  "sku": [
                    {
                      "colorHex": "#FFF",
                      "colorName": "BRANCO",
                      "colorRefId": "5678",
                      "sizes": [
                        {
                          "disabled": false,
                          "skuId": "5678",
                          "value": "P",
                        },
                      ],
                    },
                  ],
                },
                {
                  "brand": "RESERVA",
                  "categoryTree": [
                    "Camisas",
                  ],
                  "flags": [
                    {
                      "type": "savings",
                      "value": 20,
                    },
                    {
                      "type": "cashback",
                      "value": 15,
                    },
                  ],
                  "image": "https://lojausereserva.vtexassets.com/arquivos/ids/8409628-400-600",
                  "prices": {
                    "listPrice": 600.9,
                    "salePrice": 400.5,
                  },
                  "productId": "1",
                  "productLink": "https://usereserva.com/camisa/p",
                  "productName": "Camisa Polo Reserva Linho Braco",
                  "sku": [
                    {
                      "colorHex": "#FFF",
                      "colorName": "BRANCO",
                      "colorRefId": "5678",
                      "sizes": [
                        {
                          "disabled": false,
                          "skuId": "5678",
                          "value": "P",
                        },
                      ],
                    },
                  ],
                },
                {
                  "brand": "RESERVA",
                  "categoryTree": [
                    "Camisas",
                  ],
                  "flags": [
                    {
                      "type": "savings",
                      "value": 20,
                    },
                  ],
                  "image": "https://lojausereserva.vtexassets.com/arquivos/ids/8409628-400-600",
                  "prices": {
                    "listPrice": 600.9,
                    "salePrice": 400.5,
                  },
                  "productId": "1",
                  "productLink": "https://usereserva.com/camisa/p",
                  "productName": "Camisa Polo Reserva Linho Braco",
                  "sku": [
                    {
                      "colorHex": "#FFF",
                      "colorName": "BRANCO",
                      "colorRefId": "5678",
                      "sizes": [
                        {
                          "disabled": false,
                          "skuId": "5678",
                          "value": "P",
                        },
                      ],
                    },
                  ],
                },
                {
                  "brand": "RESERVA",
                  "categoryTree": [
                    "Camisas",
                  ],
                  "flags": [],
                  "image": "https://lojausereserva.vtexassets.com/arquivos/ids/8409628-400-600",
                  "prices": {
                    "listPrice": 600.9,
                    "salePrice": 0,
                  },
                  "productId": "1",
                  "productLink": "https://usereserva.com/camisa/p",
                  "productName": "Camisa Polo Reserva Linho Braco",
                  "sku": [
                    {
                      "colorHex": "#FFF",
                      "colorName": "BRANCO",
                      "colorRefId": "5678",
                      "sizes": [
                        {
                          "disabled": false,
                          "skuId": "5678",
                          "value": "P",
                        },
                      ],
                    },
                  ],
                },
              ]
            }
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={true}
            keyExtractor={[Function]}
            onContentSizeChange={[Function]}
            onLayout={[Function]}
            onMomentumScrollBegin={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            showsHorizontalScrollIndicator={false}
            stickyHeaderIndices={[]}
            viewabilityConfigCallbackPairs={[]}
          >
            <View>
              <View
                onFocusCapture={[Function]}
                onLayout={[Function]}
                style={
                  [
                    {
                      "flexDirection": "row",
                    },
                    null,
                  ]
                }
              >
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "borderRadius": 10,
                      "opacity": 1,
                      "paddingHorizontal": 5,
                      "paddingVertical": 10,
                    }
                  }
                >
                  <Image
                    source={
                      {
                        "uri": "https://lojausereserva.vtexassets.com/arquivos/ids/8409628-400-600",
                      }
                    }
                    style={
                      {
                        "borderRadius": 10,
                        "height": 270,
                        "width": 165,
                      }
                    }
                  />
                  <Text
                    style={
                      {
                        "fontFamily": "ReservaSans-Bold",
                        "fontSize": 15,
                        "lineHeight": 20,
                      }
                    }
                  >
                    Camisa Polo Reserva Li..
                  </Text>
                  <View
                    style={
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                      }
                    }
                  >
                    <Text
                      style={
                        {
                          "fontFamily": "ReservaSans-Bold",
                          "fontSize": 17,
                          "lineHeight": 23,
                        }
                      }
                    >
                      R$ 400
                    </Text>
                    <Text
                      style={
                        {
                          "fontFamily": "ReservaSans-Bold",
                          "fontSize": 11,
                          "marginTop": -4,
                        }
                      }
                    >
                      ,50
                    </Text>
                    <Text
                      style={
                        {
                          "color": "#8D8D8D",
                          "fontFamily": "Nunito-Regular",
                          "fontSize": 17,
                          "lineHeight": 23,
                          "marginLeft": 10,
                          "textDecorationLine": "line-through",
                        }
                      }
                    >
                      600
                    </Text>
                    <Text
                      style={
                        {
                          "color": "#8D8D8D",
                          "fontFamily": "Nunito-Regular",
                          "fontSize": 11,
                          "lineHeight": 17,
                        }
                      }
                    >
                      ,90
                    </Text>
                  </View>
                  <View
                    style={
                      {
                        "alignItems": "center",
                        "backgroundColor": "#000000",
                        "borderRadius": 30,
                        "flexDirection": "row",
                        "height": 30,
                        "justifyContent": "center",
                        "left": 20,
                        "position": "absolute",
                        "top": 240,
                        "width": 71,
                      }
                    }
                  >
                    <Text
                      style={
                        {
                          "color": "#ffffff",
                          "fontFamily": "WorkSans-BoldItalic",
                          "fontSize": 13,
                          "lineHeight": 15,
                        }
                      }
                    >
                      20%
                    </Text>
                    <Text
                      style={
                        {
                          "color": "#ffffff",
                          "fontFamily": "WorkSans-Italic",
                          "fontSize": 13,
                          "letterSpacing": -1.5,
                          "lineHeight": 15,
                          "marginLeft": 5,
                        }
                      }
                    >
                      OFF
                    </Text>
                  </View>
                  <View
                    style={
                      {
                        "alignItems": "center",
                        "backgroundColor": "#F2F2F2",
                        "borderRadius": 30,
                        "flexDirection": "row",
                        "height": 15,
                        "justifyContent": "center",
                        "width": 165,
                      }
                    }
                  >
                    <Text
                      style={
                        {
                          "color": "#000000",
                          "fontFamily": "WorkSans-Bold",
                          "fontSize": 12,
                          "lineHeight": 15,
                        }
                      }
                    >
                      Ganhe 15%
                    </Text>
                    <Text
                      style={
                        {
                          "color": "#000000",
                          "fontFamily": "WorkSans-Regular",
                          "fontSize": 12,
                          "lineHeight": 15,
                        }
                      }
                    >
                       de cashback
                    </Text>
                  </View>
                </View>
              </View>
              <View
                onFocusCapture={[Function]}
                onLayout={[Function]}
                style={
                  [
                    {
                      "flexDirection": "row",
                    },
                    null,
                  ]
                }
              >
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "borderRadius": 10,
                      "opacity": 1,
                      "paddingHorizontal": 5,
                      "paddingVertical": 10,
                    }
                  }
                >
                  <Image
                    source={
                      {
                        "uri": "https://lojausereserva.vtexassets.com/arquivos/ids/8409628-400-600",
                      }
                    }
                    style={
                      {
                        "borderRadius": 10,
                        "height": 270,
                        "width": 165,
                      }
                    }
                  />
                  <Text
                    style={
                      {
                        "fontFamily": "ReservaSans-Bold",
                        "fontSize": 15,
                        "lineHeight": 20,
                      }
                    }
                  >
                    Camisa Polo Reserva Li..
                  </Text>
                  <View
                    style={
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                      }
                    }
                  >
                    <Text
                      style={
                        {
                          "fontFamily": "ReservaSans-Bold",
                          "fontSize": 17,
                          "lineHeight": 23,
                        }
                      }
                    >
                      R$ 400
                    </Text>
                    <Text
                      style={
                        {
                          "fontFamily": "ReservaSans-Bold",
                          "fontSize": 11,
                          "marginTop": -4,
                        }
                      }
                    >
                      ,50
                    </Text>
                    <Text
                      style={
                        {
                          "color": "#8D8D8D",
                          "fontFamily": "Nunito-Regular",
                          "fontSize": 17,
                          "lineHeight": 23,
                          "marginLeft": 10,
                          "textDecorationLine": "line-through",
                        }
                      }
                    >
                      600
                    </Text>
                    <Text
                      style={
                        {
                          "color": "#8D8D8D",
                          "fontFamily": "Nunito-Regular",
                          "fontSize": 11,
                          "lineHeight": 17,
                        }
                      }
                    >
                      ,90
                    </Text>
                  </View>
                  <View
                    style={
                      {
                        "alignItems": "center",
                        "backgroundColor": "#000000",
                        "borderRadius": 30,
                        "flexDirection": "row",
                        "height": 30,
                        "justifyContent": "center",
                        "left": 20,
                        "position": "absolute",
                        "top": 240,
                        "width": 71,
                      }
                    }
                  >
                    <Text
                      style={
                        {
                          "color": "#ffffff",
                          "fontFamily": "WorkSans-BoldItalic",
                          "fontSize": 13,
                          "lineHeight": 15,
                        }
                      }
                    >
                      20%
                    </Text>
                    <Text
                      style={
                        {
                          "color": "#ffffff",
                          "fontFamily": "WorkSans-Italic",
                          "fontSize": 13,
                          "letterSpacing": -1.5,
                          "lineHeight": 15,
                          "marginLeft": 5,
                        }
                      }
                    >
                      OFF
                    </Text>
                  </View>
                  <View
                    style={
                      {
                        "alignItems": "center",
                        "backgroundColor": "#F2F2F2",
                        "borderRadius": 30,
                        "flexDirection": "row",
                        "height": 15,
                        "justifyContent": "center",
                        "width": 165,
                      }
                    }
                  >
                    <Text
                      style={
                        {
                          "color": "#000000",
                          "fontFamily": "WorkSans-Bold",
                          "fontSize": 12,
                          "lineHeight": 15,
                        }
                      }
                    >
                      Ganhe 15%
                    </Text>
                    <Text
                      style={
                        {
                          "color": "#000000",
                          "fontFamily": "WorkSans-Regular",
                          "fontSize": 12,
                          "lineHeight": 15,
                        }
                      }
                    >
                       de cashback
                    </Text>
                  </View>
                </View>
              </View>
              <View
                onFocusCapture={[Function]}
                onLayout={[Function]}
                style={
                  [
                    {
                      "flexDirection": "row",
                    },
                    null,
                  ]
                }
              >
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "borderRadius": 10,
                      "opacity": 1,
                      "paddingHorizontal": 5,
                      "paddingVertical": 10,
                    }
                  }
                >
                  <Image
                    source={
                      {
                        "uri": "https://lojausereserva.vtexassets.com/arquivos/ids/8409628-400-600",
                      }
                    }
                    style={
                      {
                        "borderRadius": 10,
                        "height": 270,
                        "width": 165,
                      }
                    }
                  />
                  <Text
                    style={
                      {
                        "fontFamily": "ReservaSans-Bold",
                        "fontSize": 15,
                        "lineHeight": 20,
                      }
                    }
                  >
                    Camisa Polo Reserva Li..
                  </Text>
                  <View
                    style={
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                      }
                    }
                  >
                    <Text
                      style={
                        {
                          "fontFamily": "ReservaSans-Bold",
                          "fontSize": 17,
                          "lineHeight": 23,
                        }
                      }
                    >
                      R$ 400
                    </Text>
                    <Text
                      style={
                        {
                          "fontFamily": "ReservaSans-Bold",
                          "fontSize": 11,
                          "marginTop": -4,
                        }
                      }
                    >
                      ,50
                    </Text>
                    <Text
                      style={
                        {
                          "color": "#8D8D8D",
                          "fontFamily": "Nunito-Regular",
                          "fontSize": 17,
                          "lineHeight": 23,
                          "marginLeft": 10,
                          "textDecorationLine": "line-through",
                        }
                      }
                    >
                      600
                    </Text>
                    <Text
                      style={
                        {
                          "color": "#8D8D8D",
                          "fontFamily": "Nunito-Regular",
                          "fontSize": 11,
                          "lineHeight": 17,
                        }
                      }
                    >
                      ,90
                    </Text>
                  </View>
                  <View
                    style={
                      {
                        "alignItems": "center",
                        "backgroundColor": "#000000",
                        "borderRadius": 30,
                        "flexDirection": "row",
                        "height": 30,
                        "justifyContent": "center",
                        "left": 20,
                        "position": "absolute",
                        "top": 240,
                        "width": 71,
                      }
                    }
                  >
                    <Text
                      style={
                        {
                          "color": "#ffffff",
                          "fontFamily": "WorkSans-BoldItalic",
                          "fontSize": 13,
                          "lineHeight": 15,
                        }
                      }
                    >
                      20%
                    </Text>
                    <Text
                      style={
                        {
                          "color": "#ffffff",
                          "fontFamily": "WorkSans-Italic",
                          "fontSize": 13,
                          "letterSpacing": -1.5,
                          "lineHeight": 15,
                          "marginLeft": 5,
                        }
                      }
                    >
                      OFF
                    </Text>
                  </View>
                </View>
              </View>
              <View
                onFocusCapture={[Function]}
                onLayout={[Function]}
                style={
                  [
                    {
                      "flexDirection": "row",
                    },
                    null,
                  ]
                }
              >
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "borderRadius": 10,
                      "opacity": 1,
                      "paddingHorizontal": 5,
                      "paddingVertical": 10,
                    }
                  }
                >
                  <Image
                    source={
                      {
                        "uri": "https://lojausereserva.vtexassets.com/arquivos/ids/8409628-400-600",
                      }
                    }
                    style={
                      {
                        "borderRadius": 10,
                        "height": 270,
                        "width": 165,
                      }
                    }
                  />
                  <Text
                    style={
                      {
                        "fontFamily": "ReservaSans-Bold",
                        "fontSize": 15,
                        "lineHeight": 20,
                      }
                    }
                  >
                    Camisa Polo Reserva Li..
                  </Text>
                  <View
                    style={
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                      }
                    }
                  >
                    <Text
                      style={
                        {
                          "fontFamily": "ReservaSans-Bold",
                          "fontSize": 17,
                          "lineHeight": 23,
                        }
                      }
                    >
                      R$ 600
                    </Text>
                    <Text
                      style={
                        {
                          "fontFamily": "ReservaSans-Bold",
                          "fontSize": 11,
                          "marginTop": -4,
                        }
                      }
                    >
                      ,90
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </RCTScrollView>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
